wallet/src/main/java/com/android/identity_credential/wallet/credman/CredmanPresentationActivity.kt

```diff
@@ -348,6 +226,102 @@ class CredmanPresentationActivity : FragmentActivity() {
         }
     }
 
+    /**
+     * Show the Presentment Flow and handle producing and sending the response.
+     * @param credentialId the int index of [Document] in [DocumentStore]
+     * @param dataElements the list of requested data points of a credential
+     * @param encodedSessionTranscript CBOR bytes
+     * @param readerPublicKey null by default means to generate the response for the 'openid4vp'
+     *      protocol, else if reader key is provided then generate a response for the  'preview'
+     *      protocol.
+     */
+    private fun showPresentmentFlowAndRespond(
```

Comment:
How about just returning `DeviceResponse` and having the callers do their thing, it's kinda weird to have this is gated on whether `readerPublicKey` is `null` or not (and this assumption will change in the future).

Also need whitespace before the @-param block. In fact, since it's a private function and obvious what it does, suggest to not add any kdoc at all (it's only required for public API).